cli/src/commands/shell/index.ts

```diff
@@ -1,11 +1,160 @@
-import { run } from '@xata.io/shell';
+import { Flags } from '@oclif/core';
+import { XataApiClient } from '@xata.io/client';
+import { generateFromContext } from '@xata.io/codegen';
+import chalk from 'chalk';
+import EventEmitter from 'events';
+import { createWriteStream } from 'fs';
+import fs, { mkdir } from 'fs/promises';
+import fetch from 'node-fetch';
+import { homedir, tmpdir } from 'os';
+import path, { dirname } from 'path';
+import RJSON from 'relaxed-json';
+import repl from 'repl';
+import { fileURLToPath } from 'url';
 import { BaseCommand } from '../../base.js';
+import { readAPIKey } from '../../key.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
 
 export default class Shell extends BaseCommand {
   static description = 'Open a shell to the current database and branch';
 
+  static flags = {
+    ...this.commonFlags,
+    code: Flags.string({
+      description: 'Fragment of code to be executed in the shell immediately after starting it'
+    })
+  };
+
   async run(): Promise<void> {
-    const databaseURL = await this.getDatabaseURL();
-    await run({ databaseURL });
+    const { flags } = await this.parse(Shell);
+    const apiKey = await readAPIKey();
+    if (!apiKey)
+      this.error('No API key found. Either use the XATA_API_KEY environment variable or run `xata auth login`');
+    const { protocol, host, databaseURL } = await this.getParsedDatabaseURL();
+
+    // Generate the file in the same dir than this package's code so it
+    // can import @xata.io/client
+    const tempFile = path.join(__dirname, `xata-${Date.now()}.mjs`);
+    try {
+      const { transpiled } = await generateFromContext('javascript', { apiKey, databaseURL });
+      await fs.writeFile(tempFile, transpiled);
+    } catch (err) {
+      const message = err instanceof Error ? err.message : String(err);
+      return this.error(message);
+    }
+
+    this.log(
+      `Connected to ${chalk.bold(databaseURL)}. There's a XataClient instance in the ${chalk.bold(
+        'xata'
+      )} global variable.`
+    );
+
+    const fetchApi = async (method: string, path: string, body?: any) => {
+      // TODO: Add support for staging, how?
+      const baseUrl = path.startsWith('/db') ? `${protocol}//${host}` : 'https://api.xata.io';
```

Comment:
We should probably use the helper methods of the base command here

Reply:
What do you think about https://github.com/xataio/client-ts/pull/292/commits/d8f59a1db34e6b54122c59809b4a19121cf147d4

Reply:
What would you use in this case?

Reply:
Not sure, but I would like to have ``xata login auth --staging`` or ``xata login auth --backend http://localhost:5001`` for example, and that it remembers it (in config). So in this particular case it knows through a base command helper which API to use.

Reply:
The old xata CLI uses an env variable. Maybe we could do the same. What I don't like is storing an API key without specifying the environment somehow. Maybe we can convert the `key` file into an `.ini` file? And having a section per environment?

Reply:
That would be great to be honest.